src/twisted/web/test/test_flatten.py

```diff
@@ -272,41 +272,57 @@ def test_commentEscaping(self) -> Deferred[List[bytes]]:
         SGML comment syntax is complicated and hard to use. This rule is more
         restrictive, and more compatible:
 
-        Comments start with <!-- and end with --> and never contain -- or >.
+        Comments start with <!-- and end with --> and HTML comments never contain -- or >,
+        whereas Outlook conditional comments contain their logic within brackets
+        that follow this format <!-- [if foo]>
 
         Also by XML syntax, a comment may not end with '-'.
 
         @see: U{http://www.w3.org/TR/REC-xml/#sec-comments}
         """
 
-        def verifyComment(c: bytes) -> None:
+        def verifyComment(z: bytes) -> None:
             self.assertTrue(
-                c.startswith(b"<!--"),
-                f"{c!r} does not start with the comment prefix",
+                z.startswith(b"<!--"),
+                f"{z!r} does not start with the comment prefix",
             )
             self.assertTrue(
-                c.endswith(b"-->"),
-                f"{c!r} does not end with the comment suffix",
+                z.endswith(b"-->"),
+                f"{z!r} does not end with the comment suffix",
             )
             # If it is shorter than 7, then the prefix and suffix overlap
             # illegally.
-            self.assertTrue(len(c) >= 7, f"{c!r} is too short to be a legal comment")
-            content = c[4:-3]
+            self.assertTrue(len(z) >= 7, f"{z!r} is too short to be a legal comment")
+            content = z[4:-3]
             self.assertNotIn(b"--", content)
-            self.assertNotIn(b">", content)
+            if b"[if " in content and (b"IE" in content or b"mso" in content):
+                self.assertIn(b">", content)
+                self.assertNotIn(b"&gt;", content)
+            else:
+                self.assertNotIn(b">", content)
             if content:
                 self.assertNotEqual(content[-1], b"-")
 
         results = []
-        for c in [
+        for z in [
             "",
             "foo---bar",
             "foo---bar-",
-            "foo>bar",
-            "foo-->bar",
             "----------------",
+            "foo > bar",
```

Comment:
Instead of extending an existing test, I would love to see a new test dedicated especially to this use case.

The docstring of the new test can explain what is tested, why it tests and what future developers should take into consideration to make sure no regressions are added.

Reply:
you're right, I was cutting corners, it's a bad habit that I will break.
I created a test for this particular change and added a proper docstring: [e909111](https://github.com/twisted/twisted/pull/11805/commits/e909111956155cf3dd3d4a09ea845e4647dc2548)

I also don't like one letter variables so I wrote descriptive variable names for my test function, but I left the other one as it was, didn't feel comfortable making that sort of change elsewhere